src/local_pathfinding/local_pathfinding/obstacles.py

```diff
@@ -101,7 +100,132 @@ def update_reference_point(self, reference: HelperLatLon):
+
+            self.collision_zone = collision_zone
+            return
+
+        if state_space is None:  # create a deafult one
+
+            sailbot_box = Point(
+                self.sailbot_position_latlon.longitude, self.sailbot_position_latlon.latitude
+            ).buffer(self.bbox_buffer_amount, cap_style=3, join_style=2)
+
+            waypoint_box = Point(self.next_waypoint.longitude, self.next_waypoint.latitude).buffer(
+                self.bbox_buffer_amount, cap_style=3, join_style=2
+            )
+            state_space = box(*MultiPolygon([sailbot_box, waypoint_box]).bounds)
+
+        latlon_polygons = self.all_land_data.intersection(state_space)
+
+        xy_polygons = Land._latlon_polygons_to_xy_polygons([latlon_polygons], self.reference)
+
+        collision_zone = MultiPolygon(xy_polygons)
+
+        if len(collision_zone.geoms) > 0:
+            # if collision_zone is empty, buffer(0) will return an empty Polygon
+            # buffer(0) will repair invalid/overlapping geometry
+            # otherwise we cant run .contains() on it
+            collision_zone = collision_zone.buffer(0)
+
+            if collision_zone.geom_type == "Polygon":
+                collision_zone = MultiPolygon([collision_zone])
+        else:
+            collision_zone = MultiPolygon()
+
+        self.collision_zone = collision_zone
+
+    @staticmethod
+    def _latlon_polygons_to_xy_polygons(
+        polygons: List[Polygon], reference: HelperLatLon
+    ) -> List[Polygon]:
+        """
+        Transforms a list of polygons from the global lat/lon coordinate system to the local
+        XY coordinate system.
+
+        Args:
+            polygons (List[Polygon]): List of polygons to be transformed.
+            reference (HelperLatLon): Lat and lon position of the reference point.
+
+        Returns:
+            List[Polygon]: List of transformed polygons.
+
+        Inner Functions:
+            _latlon_to_xy_point(point: HelperLatLon) -> Point:
+                Converts a latlon point to a 2D Cartesian point.
+            _latlons_to_xy_points(poly: Polygon) -> Polygon:
+                Applies the _latlon_to_point function to every point of poly
+        """
+
+        def _latlon_polygon_to_xy_polygon(poly: Polygon) -> Polygon:
+            return Polygon(list(map(_latlon_point_to_xy_point, poly.exterior.coords)))
+
+        def _latlon_point_to_xy_point(latlon_point: tuple) -> Point:
+            return Point(
+                *latlon_to_xy(
+                    reference=reference,
+                    # points are (lon, lat) in the land dataset
+                    latlon=HelperLatLon(longitude=latlon_point[0], latitude=latlon_point[1]),
+                )
+            )
+
+        if len(polygons) == 0:
+            return []
+
+        elif len(polygons) == 1:
+            return [_latlon_polygon_to_xy_polygon(polygons[0])]
+
+        return list(map(_latlon_polygon_to_xy_polygon, polygons))
```

Comment:
What's the rationale for the different returns? I'm concerned the calling code will need to be more complicated to handle these cases

Reply:
The rationale is that based on the proximity to land and the shape of the coastline and state-space, we could be dealing with multiple polygons, a single polygon, or none so the function needs to consider those three possible cases. Only `update_land_collision_zone` will call this function. I am thinking of moving this function to `coord_ssytems.py` thoughts?